src/sndfile.c

```diff
@@ -888,7 +888,7 @@ sf_command	(SNDFILE *sndfile, int command, void *data, int datasize)
 			if (data == NULL)
 			{	if (psf)
 					psf->error = SFE_BAD_COMMAND_PARAM ;
-				return SFE_BAD_COMMAND_PARAM ;
+				return 0 ;
```

Comment:
Sorry, that's not correct. Calling code should check the return of `sf_command` and only if it is non-zero should the data be trusted.

Reply:
Documentation says that `SFC_GET_LIB_VERSION returns string length.`But it is not.

The problem is mixing string length and error code at return.

Error return result was 31 (SFE_BAD_COMMAND_PARAM) before, that can be interpreted as success with string of 31 chars returned.

Now it is zero and clearly indicates that nothing was copied.

Reply:
> Documentation says that SFC_GET_LIB_VERSION returns string length.`But it is not.

The code is correct, the documentation needs to be fixed.

Reply:
On second thought, this probably is a better solution.